sql/src/main/java/com/vmware/ddlog/DDlogJooqProvider.java

```diff
@@ -362,23 +384,37 @@ private static DDlogRecord maybeOption(final Boolean isNullable, final DDlogReco
     }
 
     @Nullable
-    private static Object structToValue(final Field<?> field, final DDlogRecord record) {
-        final Class<?> cls = field.getType();
-        if (record.isStruct() && record.getStructName().equals(DDLOG_NONE)) {
-            return null;
-        }
-        if (record.isStruct() && record.getStructName().equals(DDLOG_SOME)) {
-            return structToValue(field, record.getStructField(0));
+    private static void structToValue(final Field<?> field, final DDlogRecord record, final Record jooqRecord) {
+        final boolean isStruct = record.isStruct();
+        if (isStruct) {
+            final String structName = record.getStructName();
+            if (structName.equals(DDLOG_NONE)) {
+                jooqRecord.setValue(field,null);
+                return;
+            }
+            if (structName.equals(DDLOG_SOME)) {
+                toValue(field, record.getStructField(0), jooqRecord);
+                return;
+            }
         }
+        toValue(field, record, jooqRecord);
+    }
+
+    private static void toValue(final Field<?> field, final DDlogRecord record, final Record jooqRecord) {
+        final Class<?> cls = field.getType();
```

Comment:
Reflection is bound to be very slow. This will probably dominate the other conversion costs.
You should consider caching the field types map.

Reply:
This is not using reflection. This returns a member of type Class<?> from JOOQ's Field<> type.